### PR TITLE
fix(Core/Spells): Fixed calculating LoS for dynamic objects.

### LINKED ISSUE #12080: Ground AOEs broken.
https://github.com/chromiecraft/chromiecraft/issues/3630

### What client do you play on?

enUS

### Faction

- [X] Alliance
- [X] Horde

### Content Phase:

- [X] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59
- [ ] 60

### Current Behaviour

Currently ground targeted area of effects aren't functioning correctly, particulary in areas with a Z axis or potential obstacles that block Line of Sight such as small rocks. This is impacting spells such as Flamestrike, Consecration and Explosive Trap. 

Suspect its related to the recent merge of this PR: https://github.com/azerothcore/azerothcore-wotlk/pull/11971 as this issue began around the same time that it was deployed.

Has been a pretty hot source of discussion in CC discord. 

### Expected Blizzlike Behaviour

Ground targeted area of effect skills should hit all monsters in their radius. 

### Source

_No response_

### Steps to reproduce the problem

Triager: I found one good spot, there may be many more but this is the first one I saw that's consistent.
`.go xyz -9797.036133 722.195251 35.278992 0 5.542893`
cast Flamestrike (2120) right in front of your character
cast Consecration (26573)
cast Explosive Trap (13813)

### Extra Notes

Triager: It's probably a regression of something. I wouldn't be able to say what exactly because there have been many mergers recently.

https://user-images.githubusercontent.com/100873791/174419799-180229ad-9339-4a2a-bc34-db3a4812c2bf.mp4

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/36f0a7757bde9573262f298aa1eaf6765af8295e

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-bg-item-reward](https://github.com/azerothcore/mod-bg-item-reward)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-cta-switch](https://github.com/azerothcore/mod-cta-switch)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-dmf-switch](https://github.com/azerothcore/mod-dmf-switch)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna](https://github.com/azerothcore/mod-eluna)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-low-level-rbg](https://github.com/azerothcore/mod-low-level-rbg)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-progression-system](https://github.com/azerothcore/mod-progression-system)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-rdf-expansion](https://github.com/azerothcore/mod-rdf-expansion)
- [mod-transmog](https://github.com/azerothcore/mod-transmog)
- [mod-weekend-xp](https://github.com/azerothcore/mod-weekend-xp)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

### PR DESCRIPTION
**Pull request by UltraNix**
Fixed #12080

## Issues Addressed:

- Closes #12080
- Closes https://github.com/chromiecraft/chromiecraft/issues/3630

## Tests Performed:

- Tested ingame.

## How to Test the Changes:

`.go xyz -9797.036133 722.195251 35.278992 0 5.542893`
cast Flamestrike (2120) right in front of your character
cast Consecration (26573)
cast Explosive Trap (13813)

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Gultask** [comment]
Working great!

**Review by IntelligentQuantum — APPROVED** [review]

**Comment by acidmanifesto** [comment]
waiting for the compiling testing to be completed before we hit the merge button.

**Comment by acidmanifesto** [comment]
Thank you for the PR

**Pull request merged**
